caas/kubernetes/clientconfig/k8s.go

```diff
@@ -15,15 +16,28 @@ import (
 
 var logger = loggo.GetLogger("juju.caas.kubernetes.clientconfig")
 
-// K8SClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.
-func K8SClientConfig() (*ClientConfig, error) {
+// NewClientConfig parses io.Reader to read the Kubernetes client configuration from
+// Stdin, then fallback to read from kubeconfig file if reader is nil. $KUBECONFIG
+// environment varibale or the default location.
+func NewClientConfig(reader io.Reader) (*ClientConfig, error) {
+	if reader == nil {
```

Comment:
The setting of the reader needs to happen outside this New function, in the NewClientConfigReader() method.
The passed in reader should not be nil.